src/cpp/shared/game/game.cpp

```diff
@@ -185,40 +185,37 @@ void philAI::RedistributeTroops(armyGroup *army1, armyGroup *army2, int a1, int
 }
 
 void game::SetRandomHeroArmies(int heroIdx, int isAI) {
-  int randomizedValue; // eax@1
-  int randomizedValue2;
-  int creatureTier2 = 0; // [sp+10h] [bp-8Ch]@1
-  int randomUpperBound; // [sp+14h] [bp-88h]@9
-  signed int armySlotIdx; // [sp+18h] [bp-84h]@3
-  signed int creatureTier; // [sp+18h] [bp-84h]@6
-  int randomLowerBound; // [sp+1Ch] [bp-80h]@9
-                        //__int16 creatureTypesAndQuantities[54]; // [sp+20h] [bp-7Ch]@1
-  armyGroup *pointerToArmyOfHeroIdx = &gpGame->heroes[heroIdx].army; // [sp+8Ch] [bp-10h]@1
-  bool hasTier[3]; // [sp+90h] [bp-Ch]@1
-  int TIER_ONE = 0;
-  int TIER_TWO = 1;
-  int TIER_THREE = 2;
+  int randomQuantity;
+  int randomUpperBound;
+  signed int armySlotIdx;
+  signed int creatureTier;
+  int randomLowerBound;
+  armyGroup *heroArmy = &gpGame->heroes[heroIdx].army;
+  bool hasTier[2];
+  const int TIER_ONE = 0;
+  const int TIER_TWO = 1;
+  const int LOW_QUANTITY = 1;
```

Comment:
Create a struct as follows:

struct randCreatEntry { // Pick a better name
   CREATURE_TYPE type;
   int lowQty;
   int highQty;
};

Change creatureTypesAndQuantities to be a two-dimensional array of these structures.

Then you can remove these two lines from the code, and also use shorter variable names because there's more context. Overall, your code becomes shorter.

Reply:
I'm not sure how to initialize these values this way unless I initialize each struct as a class with a constructor or I use a function to make the initializations. Is there something that I'm missing here?

Reply:
Also, I don't think CREATURE_TYPE is a valid type. It looks familiar, but I'm having trouble finding it.

Reply:
Yes, I probably meant CREATURES. I didn't check the name before writing it.

https://en.wikipedia.org/wiki/Struct_(C_programming_language)#Struct_initialization

Reply:
I'm using CREATURES here.

Reply:
I didn't see your last comment until I posted my previous one and refreshed the page.